packages/nameguard-sdk/src/index.ts

```diff
@@ -790,21 +953,92 @@ export class NameGuard {
       options.body = JSON.stringify(body);
     }
 
-    const response = await fetch(url, options);
+    try {
+      const timeoutId = setTimeout(() => {
+        this.abortController.abort();
+      }, 30000); // 30 second timeout
```

Comment:
We could make this configurable if we need.

I think we could attach timeout to the signal instead from looking at [this page](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static). I need to test it.

Reply:
I've added this to the constructor now.